### PR TITLE
KOGITO-1851- readme for management console(process list and details)

### PR DESCRIPTION
**Pull request by AjayJagan**

### CODE REVIEW

File: management-console/README.md [hunk 1]
```diff
@@ -60,4 +60,127 @@ mvn package -Dui.deps -Dui
 The native build of the application bundling in the React JS frontend does not differ from the instructions above. The only thing that's new is again the invocation of UI specific profiles.
 ```
 mvn package -Dui -Dnative
-```
\ No newline at end of file
+```
+
+## Working with management-console features
+
+### Process instance list
```
Comment (cristianonicolai) [inline management-console/README.md]:
```suggestion
### Process instances
```

File: management-console/README.md [hunk 2]
```diff
@@ -60,4 +60,127 @@ mvn package -Dui.deps -Dui
...
+
```
Comment (cristianonicolai) [inline management-console/README.md:69]: @AjayJagan perhaps a small introduction about the list itself? how it aggregates processes, the data that is shown, etc.

File: management-console/README.md [hunk 3]
```diff
@@ -60,4 +60,127 @@ mvn package -Dui.deps -Dui
...
+##### b) Filter by business key :
+  ![Businesskeysearch](./docs/businesskeysearch.gif "Businessekeysearch")
+
+  The business key is a human-readable text that might or mightnot be present in the process instance. The business key, if available would appear as a **blue coloured badge** near the process name in the list. We can enter the business key or parts of business key in the textbox and either click **Apply filter** or press **Enter** key to apply the filter. Similar to the Status filter, the chips with the typed keywords would appear below the textbox. The search for business key works on a *OR* based condition.Also to remove a filter based on the keyword, click on the '**X**' in the chip to remove and the list will reload to show the applied filter. The search supports [Wild cards](https://en.wikipedia.org/wiki/Wildcard_character "Wild cards") in its keywords. For example, consider a process having a business key as *WIOO2E*. This business key can be searched by entering *W\** or *\*OO\** or *WIOO2E*.
```
Comment (cristianonicolai) [inline management-console/README.md]:
```suggestion
  The business key is a business relevant identifier that might or might not be present in the process instance. The business key, if available would appear as a **blue coloured badge** near the process name in the list. We can enter the business key or parts of business key in the textbox and either click **Apply filter** or press **Enter** key to apply the filter. Similar to the Status filter, the chips with the typed keywords would appear below the textbox. The search for business key works on a *OR* based condition. Also to remove a filter based on the keyword, click on the '**X**' in the chip to remove and the list will reload to show the applied filter. The search supports [Wild cards](https://en.wikipedia.org/wiki/Wildcard_character "Wild cards") in its keywords. For example, consider a process having a business key as *WIOO2E*. This business key can be searched by entering *W\** or *\*OO\** or *WIOO2E*. 
```

File: management-console/README.md [hunk 4]
```diff
@@ -60,4 +60,127 @@ mvn package -Dui.deps -Dui
...
+
+##### b) Filter by business key :
+  ![Businesskeysearch](./docs/businesskeysearch.gif "Businessekeysearch")
+
+  The business key is a human-readable text that might or mightnot be present in the process instance. The business key, if available would appear as a **blue coloured badge** near the process name in the list. We can enter the business key or parts of business key in the textbox and either click **Apply filter** or press **Enter** key to apply the filter. Similar to the Status filter, the chips with the typed keywords would appear below the textbox. The search for business key works on a *OR* based condition.Also to remove a filter based on the keyword, click on the '**X**' in the chip to remove and the list will reload to show the applied filter. The search supports [Wild cards](https://en.wikipedia.org/wiki/Wildcard_character "Wild cards") in its keywords. For example, consider a process having a business key as *WIOO2E*. This business key can be searched by entering *W\** or *\*OO\** or *WIOO2E*. 
+
+#### Multi select checkbox
```
Comment (cristianonicolai) [inline management-console/README.md]:
```suggestion
#### Bulk Process Actions
```

File: management-console/README.md [hunk 5]
```diff
@@ -60,4 +60,127 @@ mvn package -Dui.deps -Dui
...
+![Multiselect](./docs/multiselect.gif "Multiselect")
+
+  The multi select is used to select multiple process instances while performing bulk [process-management](#process-management) actions.The multi select checkbox by default selects all the parent and child process instances(if loaded). It also has drop-down actions to select none (removes all selections), select only parent process instances, select all the process instances (both parent and child). The multi-select works in correlation with the **Aborted** **selected** button which is present in the top-right corner.This is a part of the [bulk operations](#bulk-operations) performed in the list
+
+In addition to these , there is a **reload** button(a sync-icon), to reload the list and a **Reset to default** button which resets all the filters to its original state(Active status and no business key filter).
+
+**Note :** all the filters are applied only to the parent process instances.
+
+#### List
```
Comment (cristianonicolai) [inline management-console/README.md]:
```suggestion
#### Process list
```

File: management-console/README.md (hunk 4 above)
Comment (cristianonicolai) [reply management-console/README.md]: perhaps bring bulk operations content to here.

File: management-console/README.md [hunk 6]
```diff
@@ -60,4 +60,127 @@ mvn package -Dui.deps -Dui
...
+  * *State* - shows the current state of the process instance
+  * *Id* - shows the full unique id of the process instance
+  * *Endpoint* - Shows the endpoint of the backend to which the client is connected
+  * *Start* - shows the time elapased from the start of the process
+  * *End* - shows the time elapased from the end of the process(if available)
+  * *Parent process* - shows if the current process instance has a parent process. Allows navigation to the parent process when clicked on. Shows the unique id when hovered over.
+  * *Sub processes* - shows if the current process instance has sub processes.Allows navigation to the respective child process when clicked on. Shows the unique id when hovered over
+
+  #### Timeline
+  The timeline shows the timeline of all the nodes of a process.It shows the state of the node(*Active* or *Completed*), the name of
```
Comment (cristianonicolai) [inline management-console/README.md]: It also shows nodes that are in error and in there the user can do retry and skip operations.

File: management-console/README.md [hunk 7]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
+
+  ![Bulkoperations](./docs/bulkoperations.gif "Bulkoperations")
+  
+  The process instance list allows the user to select multiple process instances and perform bulk process management operations.It consist of *Abort seleted*.The user can select individual process instances by checking the checkbox present in the list or use the [multi select checkbox](#multi-select-checkbox) to select the process instances.Then clicking on the *Abort selected* will open a box to show the instances being aborted and the instances being skipped(An instance which is already in *Completed* or *Aborted* cannot be Aborted again, hence the instances are skipped).
```
Comment (nmirasch) [inline management-console/README.md]: @AjayJagan *Abort seleted* --> *Abort selected

File: management-console/README.md [hunk 8]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
 ```
 mvn package -Dui -Dnative
-```
\ No newline at end of file
+```
+
+## Working with management-console features
+
+### Process instances
+The process instances page shows the list of process instances available. The list is an expandable list , which shows all the child process instance of the particular process instance when expanded. Each row in the list contains the information about the process instances. The info includes the *name* of the process instances, the *enpoint* of the process, the *status* of the process etc. The list also provides many ways to [filter](#filters) out the process instances and also to perform [process-management](#process-management) operations to the processes. The details of them are mentioned below.
```
Comment (nmirasch) [inline management-console/README.md]: "The info includes the *name* of the process instances, the *enpoint* of the process, the *status* of the process etc" -->
"The info includes process instance  details like *name*, *status*, *endpoint*, etc...

File: management-console/README.md [hunk 9]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
+
+In addition to these , there is a **reload** button(a sync-icon), to reload the list and a **Reset to default** button which resets all the filters to its original state(Active status and no business key filter).
+
+**Note :** all the filters are applied only to the parent process instances.
+
+#### Process list
+
+![Processlist](./docs/processlist.png?raw=true "Processlist")
+
+  The is shows details of the process instances. Initially it loads only the parent process instances. The following are the details shows in the list :
```
Comment (nmirasch) [inline management-console/README.md]: @AjayJagan The is --> The list?
the details shows --> the details shown

Comment (AjayJagan) [reply management-console/README.md]: The List would come :+1:

File: management-console/README.md [hunk 10]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
 ```
 mvn package -Dui -Dnative
-```
\ No newline at end of file
+```
+
+## Working with management-console features
+
+### Process instances
+The process instances page shows the list of process instances available. The list is an expandable list , which shows all the child process instance of the particular process instance when expanded. Each row in the list contains the information about the process instances. The info includes details like *name*, *status*, *endpoint* etc... The list also provides many ways to [filter](#filters) out the process instances and also to perform [process-management](#process-management) operations to the processes. The details of them are mentioned below.
```
Comment (Sara4994) [inline management-console/README.md:68]:
```suggestion
The process instances page shows the list of process instances available. The list is an expandable list , which shows all the child process instances of a particular process instance when expanded. Each row in the list contains the information about the process instances. The info includes details like *name*, *status*, *endpoint* etc... The list also provides many ways to [filter](#filters) out the process instances and also to perform [process-management](#process-management) operations to the processes. The details of them are mentioned below.
```

File: management-console/README.md [hunk 11]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
+
+#### Filters
+
+![Filters](./docs/filters.png?raw=true "Filters")
+
+The process instance list shows the list of instances available.It consists of two filters to filter the list:
```
Comment (Sara4994) [inline management-console/README.md]: "The process instance list shows the list of instances available".  -> do we need this line again as we have already talked about process instance list above. and how about using "It has two ways of filtering the list:"

File: management-console/README.md [hunk 12]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
+
+
+There are five status in total :
+ * Active
+ * Completed
+ * Aborted
+ * Suspended
+ * Error
+ 
+Initially the process instance list loads all the active instances available.To search using different filters, click on the dropdown with the **Status** tag and check the necessary status from the drowdown. Click on **Apply filter** to load the new list.Also there is a **chip array** below the filters to show the current filters applied. The filters for status can be removed by either by unchecking the status in the dropdown and clicking on **Apply filter** or clicking on the '**X**' in the chip.
```
Comment (Sara4994) [inline management-console/README.md]:
```suggestion
Initially the process instance list loads all the active instances available.To search using different filters, click on the dropdown with the **Status** tag and select the necessary status from the drowdown. Click on **Apply filter** to load the new list.Also there is a **chip array** below the filters to show the current filters applied. The filters for status can be removed by either by deselecting the options from the dropdown and clicking on **Apply filter** or by clicking on the '**X**' in the chip.
```

File: management-console/README.md [hunk 13]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
+
+  ![Bulkoperations](./docs/bulkoperations.gif "Bulkoperations")
+  
+  The process instance list allows the user to select multiple process instances and perform bulk process management operations.It consist of *Abort selected*.The user can select individual process instances by checking the checkbox present in the list or use the [multi select checkbox](#multi-select-checkbox) to select the process instances.Then clicking on the *Abort selected* will open a box to show the instances being aborted and the instances being skipped(An instance which is already in *Completed* or *Aborted* cannot be Aborted again, hence the instances are skipped).
```
Comment (Sara4994) [inline management-console/README.md]: Can we change "checking the checkbox" to "selecting the checkbox". find it more appropriate.

File: management-console/README.md [hunk 14]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
+
+  The List shows details of the process instances. Initially it loads only the parent process instances. The following are the details shows in the list :
+  * Checkbox
+  * Process name
+  * State of the process
+  * when was the process created
+  * when was the process updated
+  * a kebab button to provide [process-management](#process-management) functions
+  
+  The list has a toggle button on the left end, which when toggled would load all the child process instances of the parent process instance.
```
Comment (Sara4994) [inline management-console/README.md]:
```suggestion
  The list has a toggle button on the left end, which when toggled would load all the child process instances of that parent process instance.
```

File: management-console/README.md [hunk 15]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
+  The child process instance also has a similar sturcture as mentioned above.
+
+  **Checkbox** - A checkbox to select the process instance to perform [process-management](#process-management) operations. Checkboxes which are disabled either do not have `kogito.service.url` missing or the `process management` capability disabled.
```
Comment (Sara4994) [inline management-console/README.md]:
```suggestion
  **Checkbox** - A checkbox to select the process instance to perform [process-management](#process-management) operations. Checkboxes which are disabled either do not have `kogito.service.url` or the `process management` capability is disabled.
```

File: management-console/README.md [hunk 16]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
+
+  There are currently three process management capabilities in the process instance list.
+  * Abort
+  * Skip
+  * Retry
+
+  ##### Abort : 
+  An instance which is either in *Active*, *Error*, *Suspended* state can be aborted. Clicking on the abort would open up a box displaying the instance which was aborted.
+  #### Skip :
+  A node in *Error* state can be skipped. Clicking on the skip would open up a box displaying the success or failure state of the skip operation.
```
Comment (Sara4994) [inline management-console/README.md]:
```suggestion
  A node in *Error* state can be skipped. Clicking on the skip would open up a box displaying the success or failure status of the skip operation.
```

File: management-console/README.md [hunk 17]
```diff
@@ -60,4 +60,130 @@ mvn package -Dui.deps -Dui
...
+  #### Retry :
+  A node in *Error* state can be Retriggered. Clicking on the Retry would open up a box displaying the success or failure state of the Retry operation.
```
Comment (Sara4994) [inline management-console/README.md]:
```suggestion
  A node in *Error* state can be Retriggered. Clicking on the Retry would open up a box displaying the success or failure status of the Retry operation.
```

File: management-console/README.md (hunk 11 above)
Comment (AjayJagan) [reply management-console/README.md]: done

File: management-console/README.md (hunk 12 above)
Comment (AjayJagan) [reply management-console/README.md]: done

File: management-console/README.md (hunk 13 above)
Comment (AjayJagan) [reply management-console/README.md]: changed to selecting the checkbox

File: management-console/README.md (hunk 14 above)
Comment (AjayJagan) [reply management-console/README.md]: done

File: management-console/README.md (hunk 15 above)
Comment (AjayJagan) [reply management-console/README.md]: done

File: management-console/README.md (hunk 16 above)
Comment (AjayJagan) [reply management-console/README.md]: changed

File: management-console/README.md (hunk 17 above)
Comment (AjayJagan) [reply management-console/README.md]: changed

**Review by Sara4994 — APPROVED** [review]

**Comment by Sara4994** [comment]
@AjayJagan look good :+1:

**Review by nmirasch — APPROVED** [review]
looks good!

**Review by cristianonicolai — APPROVED** [review]

**Comment by cristianonicolai** [comment]
@AjayJagan needs rebase :)

File: management-console/README.md [hunk 18]
```diff
@@ -108,4 +107,126 @@ To check the list of processes, you may have to expand each row of the table by
 
 To get to know more about a process, you can always click on any process instance to direct to process details page and there you can see details about the process as well as the timeline and process variables.
 
-![navigate](./docs/navigate.gif "navigate")
\ No newline at end of file
+![navigate](./docs/navigate.gif "navigate")
+
+## Working with management-console features
```
Comment (cristianonicolai) [inline management-console/README.md:65]: @AjayJagan I think your content should come first, then domain explorer would be a ### Domain Explorer section

**Pull request merged**